### PR TITLE
[4657] Reactify Global App (Logging Levels)

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4657

### CODE REVIEW

File: ui/app/src/components/LoggingLevelsGrid/LoggingLevelsGridSkeletonTable.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            </GlobalAppGridCell>
+            <GlobalAppGridCell className="bordered width20">
+              <Typography variant="subtitle2">
+                <FormattedMessage id="loggingLevels.currentLevel" defaultMessage="Current Level" />
+              </Typography>
+            </GlobalAppGridCell>
+          </GlobalAppGridRow>
+        </TableHead>
+        <TableBody>
+          {items?.map((item, index) => (
```
Comment (aachavarria) [inline ui/app/src/components/LoggingLevelsGrid/LoggingLevelsGridSkeletonTable.tsx]: please remove the nullish validation '?'

File: ui/app/src/components/LoggingLevelsGrid/LoggingLevelsGridUI.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+              </GlobalAppGridCell>
+              <GlobalAppGridCell className="bordered width20">
+                <Typography variant="subtitle2">
+                  <FormattedMessage id="loggingLevels.changeLevelTo" defaultMessage="Current Level" />
+                </Typography>
+              </GlobalAppGridCell>
+            </GlobalAppGridRow>
+          </TableHead>
+          <TableBody>
+            {loggers?.map((logger) => (
```
Comment (aachavarria) [inline ui/app/src/components/LoggingLevelsGrid/LoggingLevelsGridUI.tsx]: please remove the nullish validation '?'

File: ui/app/src/components/LoggingLevelsGrid/styles.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  createStyles({
+    tableRoot: {
+      tableLayout: 'fixed'
+    },
+    select: {
+      paddingTop: '10px',
+      paddingBottom: '10px'
+    }
+  })
+);
```
Comment (aachavarria) [inline ui/app/src/components/LoggingLevelsGrid/styles.ts:29]: add export default useStyles;

Comment (rart) [reply ui/app/src/components/LoggingLevelsGrid/styles.ts:29]: (leave the export & export default below)

File: ui/app/src/components/LoggingLevelsGrid/LoggingLevelsGridUI.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+              <GlobalAppGridRow key={logger.name} className="hoverDisabled">
+                <GlobalAppGridCell align="left" className="">
```
Comment (rart) [inline ui/app/src/components/LoggingLevelsGrid/LoggingLevelsGridUI.tsx]: Remove empty className.
If copy-pasted, please remove from original location too.

File: ui/app/src/components/LoggingLevelsGrid/LoggingLevelsGridUI.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { Resource } from '../../models/Resource';
+import { Logger } from '../../models/Logger';
+import { useStyles } from './styles';
```
Comment (rart) [inline ui/app/src/components/LoggingLevelsGrid/LoggingLevelsGridUI.tsx]: use default style import

File: ui/app/src/components/LoggingLevelsManagement/LoggingLevelsManagement.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import { Logger } from '../../models/Logger';
+import ApiResponse from '../../models/ApiResponse';
+import { fetchLoggers as fetchLoggersService, setLogger } from '../../services/logs';
+import GlobalAppToolbar from '../GlobalAppToolbar';
+import { FormattedMessage } from 'react-intl';
+import { useLogicResource } from '../../utils/hooks';
+import { SuspenseWithEmptyState } from '../SystemStatus/Suspencified';
+import LoggingLevelsGridUI, { LoggingLevelsGridSkeletonTable } from '../LoggingLevelsGrid';
+
+export type LoggerLevel = 'debug' | 'warn' | 'info' | 'error';
```
Comment (rart) [inline ui/app/src/components/LoggingLevelsManagement/LoggingLevelsManagement.tsx]: This type belongs in its own file in models.
Both here and the service should employ the use of this type.

File: ui/app/src/components/LoggingLevelsManagement/LoggingLevelsManagement.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    fetchLoggers();
+  }, [fetchLoggers]);
+
+  const changeLevel = (logger: Logger, level: LoggerLevel) => {
+    logger.level = level;
+    setLogger(logger.name, level).subscribe(
+      () => {
+        fetchLoggers();
+      },
+      () => {}
```
Comment (rart) [inline ui/app/src/components/LoggingLevelsManagement/LoggingLevelsManagement.tsx]: Missing error handling.

**Comment by rart** [comment]
Merged@https://github.com/craftercms/studio-ui/pull/2302

**Pull request closed without merging**